### DISCUSSION TITLE
How to write .vtt and .srt when running whisper in python?

### DISCUSSION THREAD

**Discussion by jamiecm17** [openai/whisper#911] [Q&A]
I'm trying to export .vtt and .srt caption files. Using command line, this happens automatically. However, there is no file output when running whisper in VSCode. Is there an additional command or argument I need to pass to output .srt and .vtt files from VSCode?

**Comment by glangford** [comment]
If you want to run it in the same way as the command line then you can supply arguments by changing `launch.json`, see this Stack Overflow article.

https://stackoverflow.com/questions/57889703/in-visual-studio-code-how-to-pass-arguments-in-launch-json

If you want to see how `whisper` writes .srt files for example, see here 
https://github.com/openai/whisper/blob/a6b36ede1f060860d5676a543176a6439d91eae6/whisper/utils.py#L102

**Comment by tflax32** [comment]
Here is some block code you can try running directly in your own env. This utilizes the pysrt library to generate an srt file:

```
import whisper
import pysrt

model = whisper.load_model("base")
audio_file = ("YOUR AUDIO FILE PATH")
result = model.transcribe(audio_file, language='en')

subs = pysrt.SubRipFile()
sub_idx = 1

for i in range(len(result["segments"])):
    start_time = result["segments"][i]["start"]
    end_time = result["segments"][i]["end"]
    duration = end_time - start_time
    timestamp = f"{start_time:.3f} - {end_time:.3f}"
    text = result["segments"][i]["text"]
    
    sub = pysrt.SubRipItem(index=sub_idx, start=pysrt.SubRipTime(seconds=start_time), 
                           end=pysrt.SubRipTime(seconds=end_time), text=text)
    subs.append(sub)
    sub_idx += 1
    
subs.save("output.srt")
```

**Comment by mjysci** [comment]
You can use `whisper.utils.get_writer` to get subtitles directly from `model.transcribe` result.  

Here's [an example](https://github.com/mjysci/video2sub/blob/dfcb16b8becfefededba4524309c8e4e406fdd5a/video2sub/utils.py#L97).

**Reply by sovdevs** [reply]
sounds great but have been having import issues with this

**Reply by glangford** [reply]
@sovdevs We need more information to be able to help. Can you say which version of whisper you are using and some sample code with the import error you are getting?

**Reply by glangford** [reply]
...also, does it work outside VSCode but break if you are running in VSCode (like the original poster)

**Reply by sovdevs** [reply]
Thanks for getting back. 
openai-whisper @ git+https://github.com/openai/whisper.git@b91c907694f96a3fb9da03d4bbdc83fbcd3a40a4
VS code locally thru virtualenv. 
To write file I am currently using

````
from whisper.utils import write_srt
        with open(output_file_path, "w", encoding="utf-8") as srt_file:
              write_srt(result["segments"], file=srt_file)
````
which works well enough as an alternative.

However, in Production, I get this however

````
    from whisper.utils import write_srt
ImportError: cannot import name 'write_srt' from 'whisper.utils
`````

**Reply by glangford** [reply]
This may be the issue (from your post in the other thread); `write_srt` has been removed. You may be using two different versions of whisper locally vs production (?)

* https://github.com/openai/whisper/discussions/886#discussioncomment-6461291
